tools/ModelGeneration/Mathematica/N2HDM.m

```diff
@@ -0,0 +1,410 @@
+(* ::Package:: *)
+
+(*Import helper functions*)
+
+
+<< (NotebookDirectory[]<>"CreateModel.m")
+
+
+(* ::Chapter:: *)
+(*In this section the scalar potential is defined. As an example the N2HDM is shown*)
+
+
+(*Define higgs fields*)
+higgsbase = {\[Rho]1, \[Rho]2, \[Eta]1, \[Eta]2, \[Psi]1, \[Psi]2, \[Zeta]1, \[Zeta]2, \[Zeta]S};
+
+
+(*Assign vevs at T=0*)
+higgsvev = {0, 0, 0, 0, 0, 0, v1, v2, vs};
+
+
+(*Assign vevs at T != 0*)
+higgsvevFiniteTemp = {0, wcb, 0, 0, 0, wcp, w1, w2, ws};
+
+
+(*Replacement list for the vevs*)
+VEVRep = Table[i[[1]]->i[[2]],{i,Transpose[{higgsbase,higgsvev}]}];
+
+
+(*Replacement list set fields zero*)
+RepHiggsZero = Table[i->0,{i,higgsbase}];
+
+
+(*Define number of Higgses*)
+nHiggs = Length[higgsbase];
+
+
+(*Define input parameters of the potential*)
+InputParameters={tanbeta, vs, L1, L2, L3, L4, L5, L6, L7, L8,m12Sq};
```

Comment:
What about moving the definition of the `InputParameters` down to before l. 70 where the `ListOfDependentParameter` is derived? After the knowledge of `VHiggsGrad` it is much more straight forward to guess which `InputParameters` are feasible for the model.

Reply:
Done.